src/Repository/RecipeRepository.php

```diff
@@ -36,15 +36,47 @@ public function findByExampleField($value)
     }
     */
 
-    /*
-    public function findOneBySomeField($value): ?Recipe
+    /**
+     * @param $query
+     * @return mixed
+     */
+    public function findByTitle($query)
     {
         return $this->createQueryBuilder('r')
-            ->andWhere('r.exampleField = :val')
-            ->setParameter('val', $value)
+            ->where('r.title LIKE :query')
+            ->setParameter('query', '%' . $query . '%')
             ->getQuery()
-            ->getOneOrNullResult()
+            ->getResult()
         ;
     }
-    */
+
+    /**
+     * @param $query
+     * @param $categories
+     * @return array
+     */
+    public function findByCategory($query, $categories)
+    {
+        $results = [];
+        $recipes = [];
+
+        foreach ($categories as $category) {
+            $qb = $this->createQueryBuilder('r');
+            $recipes[] = $qb->addSelect('r')
+                ->innerJoin('r.categories', 'c')
+                ->where('r.title LIKE :query')
+                ->andWhere('c.id = :id')
+                ->setParameters([ 'query' => '%' . $query . '%', 'id' => $category->getId() ])
```

Comment:
Au lieu de faire plusieurs requêtes tu peux en faire une 
`SELECT Id, CompanyName, City, Country
  FROM Supplier
 WHERE Country IN ('USA', 'UK', 'Japan')`
->setParameter('id', array_values($categoryId)